### DISCUSSION TITLE
Why does MLflow convert conda dependencies to pip dependencies in conda.yaml

### DISCUSSION THREAD

**Discussion by slavickk** [mlflow/mlflow#6588] [Q&A] [answered]
I'm observing that the conda.yaml in mlruns/0/.../artifacts/model installs packages from pip while the original conda.yaml in the project directory installs the packages from conda. Is this a bug or a design decision? How do I preserve conda as the installation source for packages that I want to be installed from conda when serving the model? Thanks in advance!

Details:

I'm running a slightly modified sklearn_elasticnet_wide example from the tutorial - I set Python to 3.8 and a newer scikit-learn to be installed from conda rather than older pip - to make it runnable on Apple Silicon.

Here is conda.yaml in the project directory:
```
name: tutorial
channels:
  - conda-forge
dependencies:
  - python=3.8
  - scikit-learn==1.1
  - pandas==1.4.3
  - pip
  - pip:
      - mlflow>=1.0
```

And here is conda.yaml in mlruns/0/.../artifacts/model:
```
channels:
- conda-forge
dependencies:
- python=3.8.13
- pip<=22.2.2
- pip:
  - mlflow
  - cloudpickle==2.1.0
  - scikit-learn==1.1.0
name: mlflow-env
```

Why is mlflow moving scikit-learn to the pip section?

**Comment by harupy** [comment, marked as answer]
@slavickk When logging a model, did you specify `conda_env`?

```python
mlflow.sklearn.log_model(..., conda_env="conda.yaml")
```

**Reply by slavickk** [reply]
I didn't, and that was the problem. Thanks @harupy!
Relevant documentation link - https://mlflow.org/docs/latest/python_api/mlflow.sklearn.html#mlflow.sklearn.log_model
